lightning/src/ln/channel.rs

```diff
@@ -6899,6 +6998,11 @@ impl<Signer: WriteableEcdsaChannelSigner> Writeable for Channel<Signer> {
 			(31, channel_pending_event_emitted, option),
 			(35, pending_outbound_skimmed_fees, optional_vec),
 			(37, holding_cell_skimmed_fees, optional_vec),
+			(
+				if self.context.channel_state & ChannelState::WaitingForBatch as u32 != 0 { 38 } else { 39 },
+				ChannelStateInt(self.context.channel_state).into_iter().collect::<Vec<ChannelState>>(),
```

Comment:
This is meant as a migration path for when we would have another compatible/non-compatible channel state flag. This would be the first step towards having a `Vec<ChannelState>` where `ChannelState` is written as an upgradeable enum, where each variant can be even or uneven.

Reply:
If this is not necessary, I'll defer on introducing such a migration path for now.